File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:

        l, r = 0, len(nums) - 1

        while l <= r:
            mid = (l + r) // 2
            element = nums[mid]

            if element == target:
                return mid
            elif element > target:
                r = mid - 1
            else:
                l = mid + 1
        
        return l


        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int j = 0;
        for(int i = 0; i < numbers.length; i++) {
            int sum = numbers[i] + numbers[numbers.length-j-1];
            if(sum == target) {
                return new int[]{i+1,numbers.length-j};
            } else if(sum > target) {
                j++;
                i--;
            } else {}
        }
        return null;
    }
}
